src/foam/dao/AbstractFileJournal.js

```diff
@@ -454,7 +454,7 @@ try {
           PropertyInfo prop = (PropertyInfo) e.next();
           mergeProperty(oldFObject, diffFObject, prop);
         }
-        return oldFObject;
+        return diffFObject.copyFrom(oldFObject);
```

Comment:
will this not overwrite the above mergeProperty?

Reply:
No, the mergeProperty copies the new data to oldFObject then we want to return the diffFObject instead because it is the new type to write to the journal.

Reply:
This would suggest we are not writing deltas to the journal, but that's not the case.

Reply:
Ok, looking in FileJournal - yes this is correct. surprising we didn't catch this sooner.